code/game/objects/effects/decals/Cleanable/misc_cleanables.dm

```diff
@@ -141,12 +141,76 @@
 	gender = PLURAL
 	density = FALSE
 	layer = TURF_LAYER
+	plane = FLOOR_PLANE
 	icon = 'icons/effects/blood.dmi'
 	icon_state = "vomit_1"
 	random_icon_states = list("vomit_1", "vomit_2", "vomit_3", "vomit_4")
 	no_clear = TRUE
 	scoop_reagents = list("vomit" = 5)
 
+
+/obj/effect/decal/cleanable/vomit/Initialize(mapload)
+	. = ..()
+	var/turf/T = get_turf(loc)
+	gravity_check = has_gravity(src, T)
+	if(get_turf(src) != loc)
+		forceMove(loc)
+	if(!gravity_check)
+		layer = MOB_LAYER
+		plane = GAME_PLANE
+		if(prob(50))
+			animate_float(src, -1, rand(30, 120))
+		else
+			animate_levitate(src, -1, rand(30, 120))
+		icon = 'icons/effects/blood_weightless.dmi'
+
+/obj/effect/decal/cleanable/vomit/Bump(atom/A, yes)
+	. = ..()
+	if(A.density)
+		splat(A)
+
+/obj/effect/decal/cleanable/vomit/Crossed(atom/movable/AM, oldloc)
+	splat(AM)
```

Comment:
this needs to be gravity checked. Unless it's not called after splat is done? Is it?